monitor.cpp

```diff
@@ -3,36 +3,41 @@
 #include <thread>
 #include <chrono>
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
+#include <string>
+using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds, std::string;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
+bool isNormalTemparature(float temperature) {
+    return ((temperature <= 102 && temperature >= 95));
+}
+
+bool isNormalPulseRate(float pulseRate) {
+    return (pulseRate >= 60 && pulseRate <= 100);
+}
+
+bool isNormalO2Saturation(float spo2) {
+    return (spo2 >= 90);
+}
+
+void print(string message) {
+    cout << message;
     for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+        cout << "\r* " << flush;
+        sleep_for(seconds(1));
+        cout << "\r *" << flush;
+        sleep_for(seconds(1));
     }
-    return 0;
-  } else if (spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+}
+
+bool checkVital(bool (*checkFunction)(float), float value, const char* warning) {
```

Comment:
`checkVital` is a boolean function. It's good to name boolean function with some form of 'is', like `isPatientOk` or `areVitalsOk` - this will read better in the caller. Try it out